### DISCUSSION TITLE
Solving remote: 'Not enough space to resize partition'

### DISCUSSION THREAD

**Discussion by Clanceeinfinity** [Archfx/duo-de#36] [Q&A] [answered]
I followed the instructions and everything worked fine, my bootslot is A so i deleted b, worked fine. then i did .\fastboot flash system aosp-arm64-ab-gapps14.0-20240725.img and first it told me FAILED (remote: 'Not enough space to resize partition'), i looked up a video on youtube and the guy said you have to delete your actual bootslot, in my case bootslot a. I did that, now it says FAILED (remote: 'No such file or directory'). I also tried the second newest build, it resulted in the same issue.

**Comment by Clanceeinfinity** [comment]
this is also the second time i tried this, on my first try i thought i was doing something wrong so i flashed firmware again and started all over, but it doesn't seem like it worked.

**Comment by Archfx** [comment, marked as answer]
Looks like your partition order is messed up currently.

I assume your device is unlocked,  If that is the case, the following steps may work. 

```shell
./fastboot delete-logical-partition system_ext_a
./fastboot delete-logical-partition system_ext_b
./fastboot delete-logical-partition product_a
./fastboot delete-logical-partition product_b

./fastboot flash system aosp-arm64-ab-gapps14.0-20240725.img
```

Let me know the results.

**Reply by diejuse** [reply]
That worked for me too.

**Reply by ThaFacialHair** [reply]
worked for me aswell

**Reply by GUUser91** [reply]
This didn't work for me. But this tip from XDA did.
https://xdaforums.com/t/guide-unbrick-or-restore-to-oos-using-only-fastboot.4289013/pa

```
fastboot delete-logical-partition product_a
fastboot create-logical-partition product_a 5120
fastboot delete-logical-partition product_b
fastboot create-logical-partition product_b 5120
```

```
fastboot delete-logical-partition system_ext_a
fastboot create-logical-partition system_ext_a 100000
fastboot delete-logical-partition system_ext_b
fastboot create-logical-partition system_ext_b 100000
```

**Reply by ozyohthree** [reply]
+1
worked for me

**Comment by Clanceeinfinity** [comment]
what will the command ./fastboot create-logical-partition system 0 do?

will it create a virtual partition? I have more time tomorrow, ill try it out and keep you updated!

**Comment by Archfx** [comment]
I have changed the above answer, first try with the new commands, let's not touch the system partition. We just need to make enough space for the new image, so that the system partition can expand. BTW what is the exact boot partition you deleted?

**Comment by Clanceeinfinity** [comment]
i did the instructions like you put on the side, i have unlocked the bootloader before, then set up the device and enabled usb debugging again, adb reboot fastboot, fastboot devices to see if its connected properly, then i deleted system_ext, and product, then i got my bootslot, it was a, i deleted b then i tried to flash and it said there was too little space, then i deleted a aswell and then it didn't find the file / directory

do you have a telegram, maybe i could send you a video of me trying :0

also rn i reflashed firmware so i can try again

**Comment by Clanceeinfinity** [comment]
ok i tried the commands and it works now, it is booting! interestingly i lost no userdata when installing this

**Comment by Archfx** [comment]
Great!

**Comment by markyrkt** [comment]
Hi  

I came across this thread after spending nearly 5 hours installing and reinstalling to try to get this to work…

I promise I followed all your instructions - my SD1 128gb takes the flash on the system but when I type flashboot reboot - it just goes to the bootloader.  (I have also tried “fastboot reboot -w”.

I have reinstalled the recovery image several times too. 

I followed the instructions on the home page and the instructions above.   

I am not sure what I am going wrong …. 

Any pointers / help really appreciated 

thanks for all your hard work.

**Reply by markyrkt** [reply]
I’m sorry…../ IT WORKED !!!!!  I AM SO Happy !!!

it was writing on “system” not “system_a” … all good

**Comment by juapalso** [comment]
Hello, can you put the commands? I am never able to install, I always get an error, what I execute:
adb reboot fastboot
fastboot delete-logical-partition system_ext
fastboot delete-logical-partition product
fastboot delete-logical-partition system_b
fastboot delete-logical-partition system_a

./fastboot delete-logical-partition system_ext_a
./fastboot delete-logical-partition system_ext_b
./fastboot delete-logical-partition product_a
./fastboot delete-logical-partition product_b

./fastboot flash system aosp-arm64-ab-gapps14.0-20240725.img

Console:
 Sending sparse 'system' 1/15 (262112 KB)           OKAY [  8.213s]
Writing 'system'                                   FAILED (remote: 'No such file or directory')
fastboot: error: Command failed

**Reply by Clanceeinfinity** [reply]
he used usb 3 instead of 2

**Reply by adamsmithpindia** [reply]
Just use,
./fastboot delete-logical-partition product
./fastboot flash system aosp-arm64-ab-gapps14.0-20240725.im
./fastboot -w

**Reply by md7u** [reply]
**Fixing the error: FAILED remote: (no such file or directory):**
If your device has A/B Partition Scheme so you can use the following commands:
```
fastboot flash system_a your_file.img
fastboot flash system_b the_same_file.img
```

**Caution**:
Make sure that you increased **system_a** and **system_b** before the operation and their space is much larger than the system that do you want to install.

**Comment by pehuang** [comment]
Hi,
After fastboot flash system just enter recovery from fastbootd and wipe data. This would resize the user data partition I guess. Without this step my duo couldn't boot well.

**Reply by ujjwalsehgal** [reply]
I am having the exact same issue. This still didn't solve it

**Comment by Cricricrikets** [comment]
same here, i deleted both system_a and system_b

now i have this errors
```
fastboot flash system aosp-arm64-ab-gapps-15.0-20241223.img 
Sending sparse 'system' 1/15 (262112 KB)           OKAY [  6.295s]
Writing 'system'                                   FAILED (remote: 'No such file or directory')
fastboot: error: Command failed
```
and before that i had 'Not enough space to resize partition' error

surface duo (first version)

**Reply by md7u** [reply]
OMG Same!!!
...
But you can changing the file place and paste the new path...
I asked **ChatGPT** and it said you should flash the **Original Firmware** first to replace missing or damaged files resulting from previous attempts then use rhis command: 
```
fastboot erase system
```
but first you should restore deleted partitions...

**Comment by md7u** [comment]
Hi, I have Redmi 10A 64/3GB (dandelion_c3l2) and i am trying install HyperOS 2.0 (Mystic GSI) and it's failed because of system partition storage and i want to increase it's size, how can I do it and which partitions are unnecessary ?

**Reply by sinebuulo** [reply]
Hi, I wanted to know if you were able to install GSI. I have a Redmi 10 and I'm in the same situation.

**Reply by shaaakirrr** [reply]
Bro I also have redmi 10 did you find a way to install gsi. Rom ..?

**Comment by Clanceeinfinity** [comment]
wrong page, this is about the duo 1 and 2. You can try to delete both system partitions, maybe that will help. @MdX07r

**Comment by Crilind** [comment]
Hello, I hope no one can help me. I've installed the latest version of Android 15 Duo DE and I can't connect to the internet via cellular. I can receive calls and make calls myself. But the internet doesn't work, even with a different SIM card. Just in case, I live in Germany, in case that has anything to do with that.

Thanks and best regards
